src/core/annotation.js

```diff
@@ -1246,6 +1246,10 @@ class Annotation {
     return null;
   }
 
+  get hasHighlightedContent() {
```

Comment:
Give that "highlight" is also a specific type of annotations, I think we should choose another term here to disambiguate between the highlight annotation type and the text under the annotations that you're indicating here. How about `get overlaysTextContent` (also aligns with `hasTextContent`) or something similar, to indicate that the annotation type is applied over text content? (This applies to all similar names in this patch too.)